Robust_Lab.qmd

```diff
@@ -46,39 +52,199 @@ Use the `stars` data in `robustbase`. This data looks at the relationship betwee
+      theme(panel.background = element_rect(fill = "transparent"))
+    ```
+
 3.  You want to test whether the following paired samples are significantly different from one another: pre = \[22,25,17,24,16,29,20,23,19,20\], post = \[18,21,16,22,19,24,17,21,23,18\]. Often researchers would run a paired sampled t-test, but you are concerned the data does not follow a normal distribution.
 
     a.  Calculate the paired differences, that is post - pre, which will result in a vector of paired differences (pdiff0 = post - pre)
 
+    ```{r}
+    pre = c(22,25,17,24,16,29,20,23,19,20)
+    post = c(18,21,16,22,19,24,17,21,23,18)
+
+    pdiff0 = post-pre
+    ```
+
     b\. Calculate the mean of the paired differences (Xpdiff0)
 
+    ```{r}
+    Xpdiff0 = mean(pdiff0)
+    ```
+
     d\. Bootstrap b) with replacement (pdiff1) and plot the histogram with `ggplot2`.
 
+    ```{r}
+    pdiff1 = boot(pdiff0, mean_func, R = 10000)
+    ```
+
     e\. Calculate the 95% confidence intervals (BCa). What can you infer from this?
 
+    ```{r}
+    pdiff1_bca = boot.ci(pdiff1, type = "bca", conf = .95)
+    pdiff1_bca
+    ```
+
+    **The confidence interval overlaps 0: the lower bound is negative, while the upper bound is positive. The implication of the overlap is that we might fail to reject the null hypothesis (H0: there is no mean difference between the pre and post measures).**
+
     f\. Plot bootstrap mean along with 95% CIs (with `ggplot2`). Use annotate to note what the vertical lines represent.
 
+    ```{r}
+    pdiff1_mean_dist = data.frame(mean = pdiff1$t)
+
+    pdiff1_mean_dist %>% ggplot(aes(mean)) +
+      geom_histogram(bins = 30, fill = "#6A5B6E") +
+      ggtitle("Bootstrapped Mean Distribution with Confidence Intervals (BCA)") +
+      ylab("Frequency") +
+      xlab("Mean") +
+      geom_vline(xintercept = mean(pdiff1$t), color = "#84a59d", linetype = "solid", linewidth = .75) + 
+       geom_vline(xintercept = pdiff1_bca$bca[4], color = "#f28482", linetype = "dashed", linewidth = .75) +
+      geom_vline(xintercept = pdiff1_bca$bca[5], color = "#f28482", linetype = "dashed", linewidth = .75) + 
+      annotate("text", x = mean(pdiff1$t) - .25, y = max(table(pdiff1$t)) + 1000, label = "Mean", color = "#84a59d", size = 3.75) +
+      annotate("text", x = pdiff1_bca$bca[4] - .4, y = max(table(pdiff1$t)) + 1000, label = "CI (BCA)", color = "#f28482", size = 3.75) +
+      theme(panel.background = element_rect(fill = "transparent"))
+    ```
+
 4.  Pepper Joe measured the length and heat of 85 chili peppers. He wants to know if smaller peppers are hotter than longer peppers.
 
     ```{r}
     #read data.table to read in
-    chili<- read.delim("/Users/jasongeller/Documents/03-Robust_Methods/data/chillis.csv")
+    chili<- read.delim("https://raw.githubusercontent.com/jgeller112/psy504-advanced-stats/main/slides/03-Robust_Methods/data/chillis.csv")
     ```
```

Comment:
Did you forget this one?

boot.function <- function(data, indices) {
resampled_data = data[indices,]
model<-glm(HEAT~LENGTH, data=resampled_data)
return(coefficients(model))}

results <- boot(chili, boot.function, R=10000)

boot.ci(results, type="bca", R=10000)

Reply:
Whoops! Sorry, I don't know how I missed that. Just added it in!